crates/hdi/src/op/test.rs

```diff
@@ -0,0 +1,205 @@
+use super::*;
+use crate as hdi;
+use crate::test_utils::set_zome_types;
+use crate::test_utils::short_hand::*;
+use test_case::test_case;
+
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq)]
+pub struct A;
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq)]
+pub struct B;
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq)]
+pub struct C;
+
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq, Default)]
+pub struct D {
+    a: (),
+    b: (),
+}
+
+#[hdk_entry_defs(skip_hdk_extern = true)]
+#[unit_enum(UnitEntryTypes)]
+#[derive(Clone, PartialEq, Eq)]
+pub enum EntryTypes {
```

Comment:
how about using https://crates.io/crates/num_enum so we can use e.g. `EntryTypes::A.into()` where `u8` is expected. i'm not sure if this conflicts with `unit_enum` but i think it's worth a try.

Reply:
We don't associate the variants with u8s. The only valid conversion is from a variant to a [`ScopedEntryDefIndex`].

Reply:
why is `u8` used in the test function signature? i'd find it more readable to use a meaningful type and name of the numbers, hence my suggestion here.

```
(...)
#[test_case(0, 2 => matches Ok(Some(UnitEntryTypes::C)) ; "unit c")]
fn test_get_unit_entry_type(
    zome_id: u8,
    entry_type: u8,
)
(...)
```

Reply:
There's two u8s which is what a `ScopedEntryDefIndex` is made from. I'm trying to reduce the amount needed to type to write a test as there's so many tests

Reply:
that's understandable. it does mean we can't use these particular tests to demonstrate best practices. i'll try incorporating the usage of `ScopedEntryDefIndex` in the integrity-template.

Reply:
You don't need to use that type directly though

Reply:
i think it's error-prone and cumbersome to use `u8` with manual counting. any suggestions?

Reply:
I don't understand what you mean?

Reply:
Keep in mind these are unit tests on private functions. You can't use them in app code.

Reply:
Yeh there is no need for happ devs to ever directly use a `u8`. In fact it would be unsafe to.

Reply:
thanks

>  In fact it would be unsafe to.

that's exactly my concern :-)

if you don't mind i'll leave this conversation open as a reminder that i want to figure out how this relates to the integrity-template.